### PR TITLE
Misleading error when generating a report with no deployment

### PR DESCRIPTION
**Pull request by kczpl**
#### Summary

The problem is that when running local tests, the report generator errors out with a misleading message instead of explicitly stating that there is no deployment and therefore cannot generate a report. The error message should be clearer.

This pull request changes the error message when there is no ongoing deployment while generating a report. It improves error handling by checking whether the current Terraform state includes the metrics instance and errors out when it doesn't. To implement this, the `ExternalDBSettings.DriverName` configuration was changed to use PostgreSQL by default.

#### Ticket Link

--> https://mattermost.atlassian.net/browse/MM-56949
  Github issue: https://github.com/mattermost/mattermost/issues/29148

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Thank you for the PR, @kczpl! Will take a look as soon as I can :)

**Comment by kczpl** [comment]
Thank you! I've also responded in the contributors channel thread.

If my understanding of the issue or approach is incorrect, I'm ready to continue working on this, as I'm still unsure about my solution. :))

File: cmd/ltctl/report.go [hunk 1]
```diff
@@ -90,6 +90,11 @@ func RunGenerateReportCmdF(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return fmt.Errorf("could not parse output: %w", err)
 		}
+
+		if !output.HasMetrics() {
+			return fmt.Errorf("the output has no metrics, no active deployment found")
```
Comment (agarciamontoro) [inline cmd/ltctl/report.go:95]: Here we can return something slightly more useful to the user, in the lines of:
```suggestion
			return fmt.Errorf("no active deployment found, use the --prometheus-url flag if you have a local or manually deployed Prometheus server running")
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thanks, @kczpl! The code looks good, just a couple of comments:
1. If no deployer.json is configured at all, there is another error that is returned earlier. I'm thinking that we can fix this by setting the default of `TerraformStateDir` as `./ltstatus` (I don't really know why I didn't do that when I added it, to be honest). The error is the following:
```
> go run ./cmd/ltctl report generate --output=base.out --label=base "2022-10-20 08:43:39" "2022-10-20 08:44:53"
Flag --prometheus-url is not set. Defaulting to the deployment's Prometheus server...Error: failed to create terraform engine: not enough permissions to create Terraform state directory "/var/lib/mattermost-load-test-ng".
Here's some alternatives you can try:
	1. Change the TerraformStateDir setting in config/deployer.json to a directory you have permissions over (recommended).
	2. Manually create the currently configured directory "/var/lib/mattermost-load-test-ng" and change its owner to your current user.
	3. Run this and all next commands as root (not recommended).
exit status 1
```
2. It seems we're missing a newline when we return the `Flag --prometheus-url is not set. Defaulting to the deployment's Prometheus server...` error, so the new error is a bit buried Can you add it? Here's what I see:
```
> go run ./cmd/ltctl report generate --output=base.out --label=base "2022-10-20 08:43:39" "2022-10-20 08:44:53"
File base.out exists. Overwrite ? (Y/n) y
Flag --prometheus-url is not set. Defaulting to the deployment's Prometheus server...Error: the output has no metrics, no active deployment found
exit status 1
```

File: cmd/ltctl/report.go (hunk 1 above)
Comment (kczpl) [reply cmd/ltctl/report.go]: sure thing! just added that

File: cmd/ltctl/report.go [hunk 2]
```diff
@@ -90,6 +90,11 @@ func RunGenerateReportCmdF(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return fmt.Errorf("could not parse output: %w", err)
 		}
+
+		if !output.HasMetrics() {
+			return fmt.Errorf("no active deployment found, use the `--prometheus-url` flag.\nIf you have a local or manually deployed Prometheus server running")
```
Comment (agnivade) [inline cmd/ltctl/report.go:95]: No need to break the line. It's a single error message. Let the user's console break the line for us :)

File: deployment/config.go [hunk 3]
```diff
@@ -93,7 +93,8 @@ type Config struct {
 	Report                report.Config
 	// Directory under which the .terraform directory and state files are managed.
 	// It will be created if it does not exist
-	TerraformStateDir string `default:"/var/lib/mattermost-load-test-ng" validate:"notempty"`
+	// ./ltstate is the default value used when `deployer.json`` is configured at all
+	TerraformStateDir string `default:"./ltstate" validate:"notempty"`
```
Comment (agnivade) [inline deployment/config.go:97]: Fine with the change, but is this somehow related to the PR?

File: cmd/ltctl/report.go (hunk 2 above)
Comment (kczpl) [reply cmd/ltctl/report.go]: Makes sense :)) I've just pushed the removal

File: cmd/ltctl/report.go [hunk 4]
```diff
@@ -90,6 +90,11 @@ func RunGenerateReportCmdF(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return fmt.Errorf("could not parse output: %w", err)
 		}
+
+		if !output.HasMetrics() {
+			return fmt.Errorf("no active deployment found, use the `--prometheus-url` flag. If you have a local or manually deployed Prometheus server running")
```
Comment (agarciamontoro) [inline cmd/ltctl/report.go:95]: This is actually the same sentence! Sorry I wasn't clear before, but what I meant with the message is that if you have deployed Prometheus yourself (not with the tool), then you need to use the `--prometheus-url` flag. I think this is clear, but I'm open to other suggestions:
```suggestion
			return fmt.Errorf("no active deployment found, use the `--prometheus-url` flag if you have a local or manually deployed Prometheus server running")
```

File: deployment/config.go (hunk 3 above)
Comment (agarciamontoro) [inline deployment/config.go:97]: I think this comment is not needed, since the tag is clear in my opinion:
```suggestion
	TerraformStateDir string `default:"./ltstate" validate:"notempty"`
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Just a couple of nits in the comments, but other than that it looks good!

File: deployment/config.go (hunk 3 above)
Comment (agarciamontoro) [reply deployment/config.go:97]: @agnivade: Not the original goal, but to get to the correct state we wanted, we need it.

File: cmd/ltctl/report.go (hunk 4 above)
Comment (kczpl) [reply cmd/ltctl/report.go:95]: I should have removed the capital letter obviously while removing `\n`.
 In my opinion, this message is quite clear:
`no active deployment found, use the `--prometheus-url` flag if you have a local or manually deployed Prometheus server running`

File: deployment/config.go (hunk 3 above)
Comment (agnivade) [reply deployment/config.go:96]: Ah, understood!

**Review by agnivade — APPROVED** [review]
Nicely done @kczpl !

**Review by agarciamontoro — APPROVED** [review]
Thank you, @kczpl, this is great! I've already tested it, so we can get this merged :)

**Comment by kczpl** [comment]
@agarciamontoro @agnivade 
thanks guys! I'm looking forward to more open issues in this repo
